compiler/stz-auto-doc.stanza

```
@@ -11,40 +12,162 @@ defpackage stz/auto-doc :
+App is responsible for running code and producing figure at given URL
+A FIG example is:
+```stanza
+ <img src=FIG:(gen-ocdb-app, svg, view(URL, two-pin-package(0.35, 0.10, 0.14, 0.35, 0.14))) height="50">
+```
+where
+* `gen-ocdb-app` is run with command line arguments of
+  1. the base of the url to write and
+  2. the url suffix (`svg`),
+  3. the code string (e.g., `view(URL, two-pin-package(...))`) to run
+* the app (e.g., `gen-ocdb-app`) is expected to bind the variable URL to the URL command line argument so that the
+    the code can access it (e.g., `view(URL, ...)`),
+* `FIG` returns the same url plus the suffix so that it can be used in the enclosing HTML or MD image form, and
+* `gen-ocdb-app.stanza` lives in `open-components-database/utils/` and can be compiled with
+```bash
+stanza open-components-database/utils/gen-ocdb-app.stanza -o build/gen-ocdb
+```
+
+TODO:
+
+ o Move pretty-printer out to general mechanism.
+     There some tricky nuances involving precedence and operator associativity.
+     Look at the type printer in stz-type to see how precedence is handled.
+     Hopefully, can be refactored together with the pretty-printer into the same mechanism.
+ o parse-iterms is fine for now. Has some bad performance characteristics that we can revisit later.
+ o the FIG processing currently runs as part of running stanza doc,
+     this will need to be revisited so that text can be updated without rerunning all the generation programs.
+;============================================================================== <DOC>
+
+defn call-plugin (name:String, arguments:Seqable<String>) -> False|String :
+  val proc = Process(name, cat([name], arguments), PROCESS-IN, PROCESS-OUT, PROCESS-ERR)
+  ; println("WRITING HELLO")
+  ; println(input-stream(proc), "\"HELLO\"")
+  ; flush(input-stream(proc))
+  val tokens = read-line(output-stream(proc))
+  val response = match(tokens:List<Token>) :
```

What should proper operation be when the reader returns false (indicating EOF)? Seems like improper operation, maybe throw an exception?

---

done.